### DISCUSSION TITLE
feat: privacy leak - admins can view others' conversations

### DISCUSSION THREAD

**Discussion by sa-** [open-webui/open-webui#2807] [General]
**Is your feature request related to a problem? Please describe.**
Right now admins can view conversations of other users. To take privacy more seriously, it would be nice if there were an option that disabled this. 

**Describe the solution you'd like**
There are 2 possible solutions
1. Add an environment variable that disables this functionality, where admins cannot view others' conversations
2. Make it so that there doesn't have to be an admin user, and all configuration is done via environment variables or yaml files

**Describe alternatives you've considered**
-

**Additional context**
In the EU privacy is taken quite seriously, there are legal requirements that need to be met.

**Comment by tjbck** [comment]
Related https://github.com/open-webui/open-webui/discussions/1581

![image](https://github.com/open-webui/open-webui/assets/25473318/fcf1f6a1-1639-41ca-be8f-18b88a1f75ea)

Users can disable their chat history from Settings > Chats. Feel free to continue the discussion!

**Reply by sa-** [reply]
But then users don't have access to chat history anymore, which is not ideal!

Ideally users have access to their chat history but admins can't access it via the UI as well

**Reply by sa-** [reply]
Thanks again for working on this project btw, it's really really cool

**Comment by justinh-rahb** [comment]
> In the EU privacy is taken quite seriously, there are legal requirements that need to be met.

Disabling the ability to see them in the UI wouldn't make it impossible to read from the database, so I don't think EU privacy regs are going to be satisfied still. Not to mention the myriad of other ways you could intercept and log the users' inputs and API outputs. I'm not saying that the justification is invalid, but this isn't as simple as just having a switch to turn admin access off, in practical terms that's easier said than done.

In North America, privacy is not taken seriously at all.. in many institutional environments, this would not be allowed to run _without_ the ability to closely monitor user activity and even **prevent** the self-removal of information.

We have an issue of competing interests here, so whatever is decided upon we must bear in mind that we need to satisfy both sides of the coin.

**Reply by sa-** [reply]
> Disabling the ability to see them in the UI wouldn't make it impossible to read from the database, so I don't think EU privacy regs are going to be satisfied still. 

I'd like to push back against this. Access to production databases is a documented process and leaves behind a paper trail for anyone that wishes to do so. 

Allowing admins to view another user's chat is not something that leaves a paper trail so this is something that should not be allowed.

Having an environment variable that disables this access should satisfy both sides of the coin/ocean :)

**Reply by justinh-rahb** [reply]
You can forgive my ignorance of the EU laws, I stay far away from them because they're a headache I don't need to deal with. As I said originally, I'm not against having an option to disable the access through the GUI, I just don't feel that it sufficiently secures access to the level that would make regulators happy.. at some level of the stack there will always be some room for this information to be leaked. So how far must we go is the real question.

**Reply by sa-** [reply]
First of all no worries about dealing with EU privacy law. It's a whole thing but at the same time I appreciate that it exists :)

> I just don't feel that it sufficiently secures access to the level that would make regulators happy.. at some level of the stack there will always be some room for this information to be leaked. So how far must we go is the real question.

The rest of the stack can be secured by taking appropriate measures, which we have actually successfully done. We would really like to migrate to Open Web UI but this one piece of functionality is blocking us, so this is as far as we would have to go.

**Reply by justinh-rahb** [reply]
Thanks for explaining. If that's going to tick the right boxes then it sounds good to me.

**Reply by sa-** [reply]
Of course! Should this be converted back to an issue then?

**Reply by gelbzucht** [reply]
@sa merci and +1 from my (also 🇪🇺 ;-) side. We are in a similar position and this one pice seems the crucial part.

**Reply by Xyz00777** [reply]
@sa- 
> First of all no worries about dealing with EU privacy law. It's a whole thing but at the same time I appreciate that it exists :)
> 
> > I just don't feel that it sufficiently secures access to the level that would make regulators happy.. at some level of the stack there will always be some room for this information to be leaked. So how far must we go is the real question.
> 
> The rest of the stack can be secured by taking appropriate measures, which we have actually successfully done. We would really like to migrate to Open Web UI but this one piece of functionality is blocking us, so this is as far as we would have to go.

would you be able to share your secure setup? would be really interesting what you used for it, if you cant share it in detail an overview would already be nice

**Reply by sa-** [reply]
In our company we have a production environment that you can only get access to for a few hours at a time. Every time you request access you have to note the reason why you need access.

**Comment by TheMasterFX** [comment]
Same problem here. For me it would be sufficient just to be able remove the "Conversations" button, which sould be configurable e.g. via environment variable such as REAL_PRIVATE_HISTORY=1
![image](https://github.com/user-attachments/assets/909ccead-3fe9-4dbb-807c-03b9219797a4)

**Reply by cheahjs** [reply]
Already implemented with `ENABLE_ADMIN_CHAT_ACCESS=false`.

**Reply by TheMasterFX** [reply]
Oh cool, I overlooked that. Thank you

**Reply by silentoplayz** [reply]
Related merged PR, for reference: https://github.com/open-webui/open-webui/pull/4351

**Reply by ccxuy** [reply]
ENABLE_ADMIN_CHAT_ACCESS=true is not working on v5.2.0

**Reply by denispol** [reply]
> ENABLE_ADMIN_CHAT_ACCESS=true is not working on v5.2.0

Confirmed, environment variable has no effect on chat history as an admin. This is a privacy risk for companies in multiple European countries (in some countries, employees are allowed to use company systems for private reasons if it remains reasonable). It would be great if such access could at least be logged, to ensure admins' actions are logged.

**Reply by ccxuy** [reply]
Environment variable has no effect, do you mean this variable should be set in other way?

**Reply by Xyz00777** [reply]
No it just should work 🤨

**Reply by ccxuy** [reply]
Can someone fix this? This code seems to be weird:  enable_admin_chat_access and not admin then show chat button
https://github.com/open-webui/open-webui/blob/b03fc97e287f31ad07bda896143959bc4413f7d2/src/lib/components/admin/Users/UserList.svelte#L411

**Reply by ccxuy** [reply]
> No it just should work 🤨

After some tries, I realize these settings somehow affect this ENABLE_ADMIN_CHAT_ACCESS function:
      - WEBUI_SESSION_COOKIE_SAME_SITE=none
      - WEBUI_SESSION_COOKIE_SECURE=False

**Reply by Xyz00777** [reply]
this shouldnt happen... does this still happens in the newest version? maybe already fixed. otherwise we should really open an issue

**Reply by ccxuy** [reply]
I tested on v0.6.5, if I delete those two env var, the chat button shows

**Reply by recrudesce** [reply]
https://docs.openwebui.com/getting-started/env-configuration#enable_admin_chat_access

You can already do this.  Set it to `false`, not `true`.  False prevents admins from reading chats, true enables it (which is the default).  Those people above saying it doesn't work are actually enabling the ability to read the chats rather than disabling it cos they didn't read the docs :P

**Comment by danielrosehill** [comment]
I just want to add to this thread (and without wishing to be negative):

The reason that this is problematic extends way beyond the realm of compliance, and I think that that's probably a less important point, given that there may be many reasons why, from a compliance standpoint, OpenWebUI wouldn't satisfy requirements. 

I have an instance and would love to introduce my wife to it as a co-user. However, that would require giving admin permissions. Similarly, I've set up an instance for use at work and would love to invite my co-workers. 

In both cases, I absolutely do not want to have the ability to view other people's private messages, nor would I wish them to be able to view mine. This isn't due to compliance, it's simply due to wishing to respect the privacy of others. 

Given that privacy is what draws many of us to use tools like this, I really hope that we can implement a system that ensures rigorous safeguards for users' conversation histories. The default position in my opinion should be that no data is shared between users unless explicitly authorised.

**Reply by TomTom101** [reply]
I would love the messages database table to be encrypted with a user password or otherwise generated key.

**Comment by Espinaydelfin** [comment]
In certain contexts, such as corporate internal use with sensitive documents, administrators must be able to conduct audits of chat activity and perform backups, among other tasks. Conversely, in private environments, it is essential to maintain user confidentiality. To achieve this, I propose that the administration panel allow configuration by groups, clearly differentiating between types of user groups and enabling private user groups to maintain encrypted data without easy administrator access. In contrast, other users would be categorized separately. Furthermore, considering the significant regulatory differences between countries regarding access controls, it is crucial to enable regional configuration from the administration panel (as some countries impose restrictions while others mandate access). From my perspective, managing this functionality through user groups, with assigned regions and corresponding privacy policies, is a more efficient approach.

**Reply by Xyz00777** [reply]
the admin can always find a way to access the data if really needed. the better question is how difficult it should be in my opinion. like in germany we have really good privacy regulations compared to in example america. but i see no point in making a difference options for both countrys if in america it just could get deactivated. and in germany admins also probably could need access in a workplace security issue where this chats also could be needed to be investigated. so its nice to be able to activate it but in example deactivated per default

**Comment by rgaricano** [comment]
Yes, similar as cryptpad do, if it is also encrypted, anonymization is avoided. Wei could have owndata or that of third parties under legal personal database protection (in spain you have  to comply with privacy protection rights and if you have a database with personal data you have to protect it with regulated standards). 
Something that is already taken for granted in some areas, such as the legal sector, where RAG is essential and data is highly sensitive, which is why local and autonomous solutions like this are sought, encryption also provides protection against unauthorized data extraction.

**Comment by C2418M31** [comment]
This is difficult to adopt at an organizational level. Even as an admin, I’m hesitant to use it, certain words or sensitive keywords tied to privileged documents should not be visible to admins, as this poses a risk of unintentional leaks.
